### PR TITLE
chore: upgrade to yarn 3

### PR DESCRIPTION
**Pull request by Nodonisko**
Migration to Yarn 3

TODO:

- [x] Remove legacy scripts for checking immutability of lockfile (use --immutable arg instead)
- [x] All executable binaries should be called using `yarn run` (aka `electron` => `yarn electron`) etc. (Probably not necessary but recommended by Yarn, also it could fix some issue that we had with executables on Nix)
- [x] Migrate all CLI commands in CI to Yarn 3 (it will fix Gitlab)
- [x] Remove deduplicate
- [x] Cache `.yarn/cache` folder => [guide here](https://github.com/actions/cache/blob/main/examples.md#node---yarn-2)

Followups:
- [ ] Remove `lerna` it could be replaced by one command from Yarn and later that should be replaced by Nx
- [ ] migrate patching to Yarn built-in system of patching https://yarnpkg.com/cli/patch/#gatsby-focus-wrapper
- [ ] Consider using `yarn workspace focus` for some scripts in Gitlab, should be faster (but maybe insignificant with caching enabled)

### CODE REVIEW

**Comment by dahaca** [comment]
Looking amazing! 🤟 💘

**Comment by vdovhanych** [comment]
Pushed a change for nix-shell builds to test mac os app if it builds with changes

File: ci/build.yml [hunk 1]
```diff
@@ -37,7 +37,7 @@ suite-web build dev:
   variables:
     ASSET_PREFIX: /suite-web/${CI_BUILD_REF_NAME}/web
   script:
-    - yarn install --frozen-lockfile --cache-folder .yarn --prefer-offline
+    - yarn install --immutable
```
Comment (matejkriz) [inline ci/build.yml:40]: what about `--immutable-cache --check-cache` params? wouldn't it be safer with them?

Comment (Nodonisko) [reply ci/build.yml:40]: Check cache will always fetch all packages from remote servers so there will be no speed up benefits. Also I think this option is needed only when you are using zero install a commiting your yarn/cache folder to git and we don't do that.

Comment (matejkriz) [reply ci/build.yml:40]: Oh, you're right, that's valid for zero install, thanks.

**Review by vdovhanych — APPROVED** [review]

**Pull request merged**